modules/version-checker/version_checker_test.go

```diff
@@ -0,0 +1,149 @@
+package version_checker
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestParamValidation(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		param                CheckVersionParams
+		containError         bool
+		expectedErrorMessage string
+	}{
+		{
+			param:                CheckVersionParams{},
+			containError:         true,
+			expectedErrorMessage: "set ExpectedVersion in params",
+		},
+		{
+			param: CheckVersionParams{
+				Binary:          -1,
+				ExpectedVersion: "1.2",
+				WorkingDir:      ".",
+			},
+			containError:         true,
+			expectedErrorMessage: "set Binary in params",
+		},
+		{
+			param: CheckVersionParams{
+				Binary:          Docker,
+				ExpectedVersion: "1.2",
+				WorkingDir:      "",
+			},
+			containError:         true,
+			expectedErrorMessage: "set WorkingDir in params",
+		},
+		{
+			param: CheckVersionParams{
+				Binary:          Docker,
+				ExpectedVersion: "abc",
+				WorkingDir:      ".",
+			},
+			containError:         true,
+			expectedErrorMessage: "invalid version format found {abc}",
+		},
+		{
+			param: CheckVersionParams{
+				Binary:          Docker,
+				ExpectedVersion: "1.2.3",
+				WorkingDir:      ".",
+			},
+			containError:         false,
+			expectedErrorMessage: "",
+		},
+	}
+
+	for _, tc := range tests {
+		err := validateParams(tc.param)
+		testCaseContextStr := fmt.Sprintf("test case w/ param: %v", tc.param)
+		if tc.containError {
+			require.EqualError(t, err, tc.expectedErrorMessage, testCaseContextStr)
+		} else {
+			require.NoError(t, err, testCaseContextStr)
+		}
+	}
+}
+
+func TestExtractVersionFromShellCommandOutput(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		outputStr            string
+		expectedVersionStr   string
+		containError         bool
+		expectedErrorMessage string
+	}{
+		{outputStr: "version is 1.2.3", expectedVersionStr: "1.2.3", containError: false,
+			expectedErrorMessage: ""},
+		{outputStr: "version is v1.0.0", expectedVersionStr: "1.0.0", containError: false,
+			expectedErrorMessage: ""},
+		{outputStr: "version is v1.0", expectedVersionStr: "1.0", containError: false,
+			expectedErrorMessage: ""},
+		{outputStr: "version is vabc", expectedVersionStr: "", containError: true,
+			expectedErrorMessage: "failed to find version using regex matcher"},
+		{outputStr: "", expectedVersionStr: "", containError: true,
+			expectedErrorMessage: "failed to find version using regex matcher"},
+	}
+
+	for _, tc := range tests {
+		versionStr, err := extractVersionFromShellCommandOutput(tc.outputStr)
+		testCaseContextStr := fmt.Sprintf("test case w/ outputStr: %s", tc.outputStr)
+		if tc.containError {
+			require.EqualError(t, err, tc.expectedErrorMessage, testCaseContextStr)
+		} else {
+			require.NoError(t, err, testCaseContextStr)
+			require.Equal(t, tc.expectedVersionStr, versionStr, testCaseContextStr)
+		}
+	}
+}
+
+func TestValidateBinaryVersionGreaterOrEqual(t *testing.T) {
```

Comment:
NIT ( so, feel free to ignore or address later as a nice-to-have, or even defer by filing a GitHub issue for): for even greater legibility, you could optionally extend all your test cases with a `Name` field that could contain a human-legible string explaining what that test case is checking for / ensuring

Then, you still loop through your defined test cases, but also leverage `t.Run`, passing the test case name as the first param, as demonstrated in [this article](https://medium.com/code-chasm/unit-test-in-go-8d17d3910818): 

```golang
for _, tt := range tests {
  t.Run(tt.Name, func(t *testing.T) {
   ...
  }
}
```

This way, both in the source code, and in the logs when we're looking at CI/CD failures, we'll be able to quickly see which cases we just broke with a recent change.

Reply:
Yeah SGTM. Updated.